src/norm/pool.nim

```diff
@@ -0,0 +1,69 @@
+import std/locks
+
+import sqlite
+
+
+type 
+  Pool* = ref object 
+    defaultSize: Natural
+    conns: seq[DbConn]
+    poolExhaustedPolicy: PoolExhaustedPolicy
+    lock: Lock
+  PoolExhaustedError* = object of CatchableError
+  PoolExhaustedPolicy* = enum
+    pepRaise
+    pepExtend
+
+
+func newPool*(defaultSize: Positive, poolExhaustedPolicy = pepRaise): Pool =
```

Comment:
I like that this goes with nims default naming convention of "new" for ref objects and "init" for value objects!
Good catch, totally overlooked that in tinypool